File: testdata/fixtures/elixir/CWE-79.ex
# CWE-79: XSS via template
defmodule Fixture.CWE79 do
  def vuln(conn, user) do
    # ruleid: elixir-xss
    Phoenix.Controller.html(conn, user)
    # ruleid: elixir-xss
    EEx.eval_string(user)
  end
  def vuln2(conn, params) do
    # ruleid: elixir-xss
    Phoenix.Controller.html(conn, params["html"])
    # ruleid: elixir-xss
    EEx.eval_string(params["template"])
  end
  def safe(conn) do
    # ok: elixir-xss
    Phoenix.Controller.html(conn, "<p>hello</p>")
    # ok: elixir-xss
    EEx.eval_string("fixed")
  end
  def safe2(conn) do
    # ok: elixir-xss
    Phoenix.Controller.html(conn, "<div>safe</div>")
  end
end

File: testdata/fixtures/clojure/CWE-79.clj
;; CWE-79: XSS
(ns fixture.cwe79)
(defn vuln [user]
  ;; ruleid: clojure-xss
  (hiccup.core/html user)
  ;; ruleid: clojure-xss
  (selmer.parser/render "t.html" user)
  ;; ok: clojure-xss
  (hiccup.core/html "hello")
  ;; ok: clojure-xss
  (selmer.parser/render "t.html" "fixed"))
